devel/expect/Portfile

```diff
@@ -74,7 +74,13 @@ configure.args      --mandir=${prefix}/share/man \
 test.run        yes
 test.target     test
 
-post-destroot       { ln -s expect${version}/libexpect${version}.a ${destroot}${prefix}/lib/libexpect.a
+post-destroot       { 
+            ln -s expect${version}/libexpect${version}.dylib ${destroot}${prefix}/lib/libexpect.dylib
+            # Fix libexpect's install name to match its new location.
+            system "install_name_tool -id ${prefix}/lib/libexpect.dylib ${destroot}${prefix}/lib/libexpect.dylib"
+            # The expect binary has already been linked against the old install
+            # name, so we need to fix that too.
+            system "install_name_tool -change libexpect${version}.dylib ${prefix}/lib/libexpect.dylib ${destroot}${prefix}/bin/expect"
```

Comment:
Instead of running `install_name_tool` afterward to fix the wrong install name, can the correct install name be set to begin with?

Reply:
I tried to do this but wasn't able to figure it out. How does one typically do this for an arbitrary port? expect doesn't seem to know about install names at all, and it uses the same flags for compiling the library as it does the actual code:

```m4
# TEA specific: use LDFLAGS_DEFAULT instead of LDFLAGS
SHLIB_LD='${CC} -dynamiclib ${CFLAGS} ${LDFLAGS_DEFAULT}'
AC_CACHE_CHECK([if ld accepts -single_module flag], tcl_cv_ld_single_module, [
hold_ldflags=$LDFLAGS
LDFLAGS="$LDFLAGS -dynamiclib -Wl,-single_module"
AC_TRY_LINK(, [int i;], tcl_cv_ld_single_module=yes, tcl_cv_ld_single_module=no)
LDFLAGS=$hold_ldflags])
AS_IF([test $tcl_cv_ld_single_module = yes], [
SHLIB_LD="${SHLIB_LD} -Wl,-single_module"
])
# TEA specific: link shlib with current and compatiblity version flags
vers=`echo ${PACKAGE_VERSION} | sed -e 's/^\([[0-9]]\{1,5\}\)\(\(\.[[0-9]]\{1,3\}\)\{0,2\}\).*$/\1\2/p' -e d`
SHLIB_LD="${SHLIB_LD} -current_version ${vers:-0} -compatibility_version ${vers:-0}"
SHLIB_SUFFIX=".dylib"
# Don't use -prebind when building for Mac OS X 10.4 or later only:
AS_IF([test "`echo "${MACOSX_DEPLOYMENT_TARGET}" | awk -F '10\\.' '{print int([$]2)}'`" -lt 4 -a \
"`echo "${CPPFLAGS}" | awk -F '-mmacosx-version-min=10\\.' '{print int([$]2)}'`" -lt 4], [
LDFLAGS="$LDFLAGS -prebind"])
LDFLAGS="$LDFLAGS -headerpad_max_install_names"
AC_CACHE_CHECK([if ld accepts -search_paths_first flag],
    tcl_cv_ld_search_paths_first, [
hold_ldflags=$LDFLAGS
LDFLAGS="$LDFLAGS -Wl,-search_paths_first"
AC_TRY_LINK(, [int i;], tcl_cv_ld_search_paths_first=yes,
	tcl_cv_ld_search_paths_first=no)
LDFLAGS=$hold_ldflags])
AS_IF([test $tcl_cv_ld_search_paths_first = yes], [
LDFLAGS="$LDFLAGS -Wl,-search_paths_first"
])
```

Should we patch this file instead?

Reply:
Oh, they're using TEA. That explains a lot and makes sense given the nature of expect. Unfortunately it's aimed at making extensions that Tcl can load, not libraries intended to be linked against by other programs, so some patching is probably unavoidable if we want it to make the latter correctly.

Reply:
> How does one typically do this for an arbitrary port?

Wherever the command is that creates the dynamic library using the `-dynamiclib` (or `-shared`) flag, it should also use the flags `-current_version`, `-compatibility_version`, and `-install_name` set to the correct values.

Reply:
Right, but the build uses the same compiler flags for building objects as it does for the shared library. So I don't see a way to inject those flags in only for the shared library generation without patching their code. I could do that, but I would ask if that is preferred to using `install_name_tool`.

Reply:
That sounds like a design flaw in the project's build system that needs to be reported to them.

Reply:
I am happy to follow up with them, but until then what should I do for this PR?

Reply:
Filed as https://core.tcl-lang.org/expect/tktview/30e2a5645dcb4538afd5659e3091dabf9cc78a5a